yt_dlp/extractor/gedidigital.py

```diff
@@ -1,198 +1,104 @@
-        'only_matching': True,
-    }, {
-        'url': 'https://video.tribunatreviso.gelocal.it/dossier/coronavirus-in-veneto/ecco-le-prima-vaccinazioni-di-massa-nella-marca/134485/135024',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.nuovavenezia.gelocal.it/locale/camion-troppo-alto-per-il-ponte-ferroviario-perde-il-carico/135734/136266',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.gazzettadimodena.gelocal.it/locale/modena-scoperta-la-proteina-che-predice-il-livello-di-gravita-del-covid/139109/139796',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.lanuovaferrara.gelocal.it/locale/due-bombole-di-gpl-aperte-e-abbandonate-i-vigili-bruciano-il-gas/134391/134957',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.corrierealpi.gelocal.it/dossier/cortina-2021-i-mondiali-di-sci-alpino/mondiali-di-sci-il-timelapse-sulla-splendida-olympia/133760/134331',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.lasentinella.gelocal.it/locale/vestigne-centra-un-auto-e-si-ribalta/138931/139466',
-        'only_matching': True,
-    }, {
-        'url': 'https://video.espresso.repubblica.it/tutti-i-video/01-ted-villa/14772',
-        'only_matching': True,
     }]
 
-    @staticmethod
-    def _sanitize_urls(urls):
-        # add protocol if missing
-        for i, e in enumerate(urls):
-            if e.startswith('//'):
-                urls[i] = f'https:{e}'
-        # clean iframes urls
-        for i, e in enumerate(urls):
-            urls[i] = urljoin(base_url(e), url_basename(e))
-        return urls
-
-    @classmethod
-    def _extract_embed_urls(cls, url, webpage):
-        return cls._sanitize_urls(tuple(super()._extract_embed_urls(url, webpage)))
-
-    @staticmethod
-    def _clean_formats(formats):
-        format_urls = set()
-        clean_formats = []
-        for f in formats:
-            if f['url'] not in format_urls:
-                if f.get('audio_ext') != 'none' and not f.get('acodec'):
-                    continue
-                format_urls.add(f['url'])
-                clean_formats.append(f)
-        formats[:] = clean_formats
-
     def _real_extract(self, url):
-        video_id, url = self._match_valid_url(url).group('id', 'base_url')
+        video_id, slug = self._match_valid_url(url).group('id', 'slug')
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_meta(
-            ['twitter:title', 'og:title'], webpage, fatal=True)
-        player_data = re.findall(
-            r"PlayerFactory\.setParam\('(?P<type>format|param)',\s*'(?P<name>[^']+)',\s*'(?P<val>[^']+)'\);",
-            webpage)
-
+        data = self._search_json(
+            r'BrightcoveVideoPlayerOptions\s*=', webpage, 'Brightcove video player options', video_id, transform_source=js_to_json,
+        )
+        streams = json.loads(data['videoSrc'])
```

Comment:
either use `self._parse_json()` if you want this to be fatal (so we get a proper error) or else inline this below if you want it to be non-fatal

e.g.
```py
        streams = self._parse_json(data['videoSrc'], video_id)
```

or

```py
        for stream in traverse_obj(data, ('videoSrc', {json.loads}, lambda _, v: url_or_none(v['src']))):
```

---

also, since these are Brightcove player options, are you sure that there is always a `videoSrc` instead of a brightcove video ID? (there is a Brightcove account ID and player ID in the options)

Reply:
Yes, never found any different option. Will make sure to test with many videos to be sure